app/src/main/java/com/duckduckgo/app/tabs/model/TabDataRepository.kt

```diff
@@ -86,7 +88,12 @@ class TabDataRepository @Inject constructor(
     }
 
     override suspend fun selectByUrlOrNewTab(url: String) {
-        val tabId = tabsDao.selectTabByUrl(url)
+        var searchUrl = url
+        if (useOurAppDetector.isUseOurAppUrl(url)) {
+            searchUrl = "%${UseOurAppDetector.USE_OUR_APP_DOMAIN}%"
```

Comment:
I've changed the variable names a bit so it directly says we expect a query rather than a URL and I've also moved the actual query `%DOMAIN_HERE%` to a const with a more decent name. Hopefully that makes it clearer. Changing the actual query to `select tabId from tabs where url = :url` won't crash the app but it wouldn't work. Now that the query is `select tabId from tabs where url LIKE :query` should make more sense.